packages/python/vc_init.py

```diff
@@ -191,23 +191,33 @@ def __init__(self, scope):
                 self.state = ASGICycleState.REQUEST
                 self.app_queue = None
                 self.response = {}
+                self._legacy_asyncio = sys.version_info < (3, 10)
 
             def __call__(self, app, body):
                 """
                 Receives the application and any body included in the request, then builds the
                 ASGI instance using the connection scope.
                 Runs until the response is completely read from the application.
                 """
-                loop = asyncio.new_event_loop()
-                self.app_queue = asyncio.Queue(loop=loop)
+                if self._legacy_asyncio:
+                    loop = asyncio.new_event_loop()
+                    self.app_queue = asyncio.Queue(loop=loop)
+                else:
+                    self.app_queue = asyncio.Queue()
                 self.put_message({'type': 'http.request', 'body': body, 'more_body': False})
 
                 asgi_instance = app(self.scope, self.receive, self.send)
 
-                asgi_task = loop.create_task(asgi_instance)
-                loop.run_until_complete(asgi_task)
+                if self._legacy_asyncio:
+                    asgi_task = loop.create_task(asgi_instance)
+                    loop.run_until_complete(asgi_task)
+                else:
+                    asyncio.run(self.run_asgi_instance(asgi_instance))
```

Comment:
Do we actually need a separate function? From what I see in the handler, at this point in time, we already know that `app` (or `app.__call__`) is a coroutine. Can't we just run it directly instead?

```suggestion
                    asyncio.run(asgi_instance)
```